spec/lib/mini_magick/shell_spec.rb

```diff
@@ -87,6 +87,19 @@
       expect(stdout).to match("my value")
     end
 
+    it "executes the command with only the environment variables from MiniMagick.cli_env (plus some required env) when MiniMagick.restricted_env is true" do
+      allow(MiniMagick).to receive(:restricted_env).and_return(true)
+      allow(MiniMagick).to receive(:cli_env).and_return({"PATH" => "/usr/bin"})
+
+      ENV['SOMETHING_SECRET'] = 'abc123'
+
+      stdout, * = subject.execute(['echo "SOMETHING_SECRET=$SOMETHING_SECRET HOME=$HOME PATH=$PATH"'])
+
+      expect(stdout).to eq("SOMETHING_SECRET= HOME=#{Dir.home} PATH=/usr/bin\n")
+    ensure
+      ENV.delete('SOMETHING_SECRET')
```

Comment:
Ah I see this failing in Ruby 2.4 and below. Is there a different way to unset env for a test that works for Ruby 2.3 and 2.4?

I usually avoid `stub_const('ENV', ...)` or `allow(ENV).to receive` because they fundamentally change the type of object ENV becomes, which has caused me test pain in the past in my other projects. But if it's more sensible for this project, I don't mind making the change =)